### PR TITLE
feat: Add segmentSpecificConfiguration and add outlineOpacity config for Segmentation

### PR DESCRIPTION
**Pull request by sedghi**
### API Change
- getToolGroupsWithSegmentation -> getToolGroupIdsWithSegmentation to better represent what the function returns

### Fix
- Volume viewport initial acquisition orientation bug 

### Improvements:
- Upgrade vtk.js dependency to 25.8.6 and fix visual tests
- Addition of referenceId to actor entries in volumeViewport for segmentations to keep track of referencedVolumeId
- Addition of `CAMERA_RESET` event for consumers to subscribe to
- Addition of `VOLUME_VIEWPORT_NEW_VOLUME` event for consumers
- `createAndCacheDerivedVolume` can now accept a target sharedArrayBuffer to create the derived volume
- Addition of a new level of configuration for segmentations: `segmentSpecificConfiguration` that let you apply a specific configuration (fillAlpha etc.) to specific segments in a segmentation
- Addition of `setToolMode` on toolGroups to be able to set a tool Active, etc. without grabbing its instance in the toolGroup
- Improved Crosshairs internals to not break when a viewport's image change
- Change Segmentations BlendMode to `MAXIMUM_INTENSITY_BLEND` to achieve a linear opacity behvaior upon config change
- Addition of `OutlineOpacity` as a configuration for segmentation

### CODE REVIEW

File: common/reviews/api/core.api.md [hunk 1]
```diff
@@ -394,6 +396,9 @@ interface CustomEvent_2<T = any> extends Event {
     initCustomEvent(typeArg: string, canBubbleArg: boolean, cancelableArg: boolean, detailArg: T): void;
 }
 
+// @public (undocumented)
+const _default: any;
```
Comment (swederik) [inline common/reviews/api/core.api.md:400]: what is this?

File: packages/core/src/RenderingEngine/VolumeViewport.ts [hunk 2]
```diff
@@ -443,11 +436,14 @@ class VolumeViewport extends Viewport implements IVolumeViewport {
     const actorEntry = this.getDefaultActor();
 
     if (!actorEntry) {
-      throw new Error(
-        'Cannot get acquisition plane orientation, no actor in viewport'
-      );
+      return;
     }
 
+    // throw new Error(
```
Comment (swederik) [inline packages/core/src/RenderingEngine/VolumeViewport.ts:442]: You can remove this

File: packages/tools/src/stateManagement/segmentation/removeSegmentationsFromToolGroup.ts [hunk 3]
```diff
@@ -55,13 +53,14 @@ function removeSegmentationsFromToolGroup(
   }
 
   segRepresentationUIDsToRemove.forEach((segmentationDataUID) => {
-    _removeSegmentation(toolGroupId, segmentationDataUID);
+    _removeSegmentation(toolGroupId, segmentationDataUID, immediate);
```
Comment (swederik) [inline packages/tools/src/stateManagement/segmentation/removeSegmentationsFromToolGroup.ts:56]: Why would you not want to do it immediately? What does this mean?

File: utils/ExampleRunner/example-info.json [hunk 4]
```diff
@@ -142,6 +142,10 @@
         "name": "ToolGroup Specific Segmentation Configuration",
         "description": "Demonstrate how to change the configuration of how a specific tool group displays segmentations through via segmentation representations"
       },
+      "segmentSpecificConfiguration": {
+        "name": "Segment Specific Segmentation Configuration",
```
Comment (swederik) [inline utils/ExampleRunner/example-info.json:146]: Segment-specific

File: yarn.lock [hunk 5]
```diff
@@ -3017,22 +3017,41 @@
     "@jridgewell/resolve-uri" "^3.0.3"
     "@jridgewell/sourcemap-codec" "^1.4.10"
 
-"@kitware/vtk.js@24.18.7":
-  version "24.18.7"
-  resolved "https://registry.yarnpkg.com/@kitware/vtk.js/-/vtk.js-24.18.7.tgz#746cb26e4c98e86f70eb51772d2407432f53e672"
-  integrity sha512-I40IUDZ9menwJtOvYKsp+FoHWTlkk7JYf2BcBCq87ayEC4hd1+wTEhXq38LjwEvCeALEaFnjrtYYXvaKt8Nwtg==
+"@kitware/vtk.js@25.8.6":
```
Comment (swederik) [inline yarn.lock:3020]: Why are there two versions? Pin it

File: packages/tools/src/tools/displayTools/SegmentationDisplayTool.ts [hunk 6]
```diff
@@ -124,25 +131,32 @@ class SegmentationDisplayTool extends BaseTool {
...
+        }
+
+        return viewportsRenderList;
       }
     );
 
-    // for all viewports in the toolGroup trigger a re-render
-    toolGroupViewports.forEach((viewport) => {
-      viewport.render();
+    Promise.all(segmentationRenderList).then(() => {
```
Comment (swederik) [inline packages/tools/src/tools/displayTools/SegmentationDisplayTool.ts:155]: What if one fails? Can that happen?

File: packages/tools/src/tools/displayTools/Labelmap/labelmapDisplay.ts [hunk 7]
```diff
@@ -117,7 +109,8 @@ async function addSegmentationRepresentation(
  */
 function removeSegmentationRepresentation(
   toolGroupId: string,
-  segmentationRepresentationUID: string
+  segmentationRepresentationUID: string,
+  immediate = false
```
Comment (swederik) [inline packages/tools/src/tools/displayTools/Labelmap/labelmapDisplay.ts:113]: If this is about rendering let's make that clear in the parameter name, like renderImmediately

File: packages/tools/src/tools/displayTools/Labelmap/labelmapDisplay.ts [hunk 8]
```diff
@@ -195,77 +208,94 @@ function _setLabelmapColorAndOpacity(
...
+
+  ofun.setClamping(false);
+  volumeActor.getProperty().setScalarOpacity(0, ofun);
+
   volumeActor.getProperty().setInterpolationTypeToNearest();
 
-  volumeActor.getProperty().setUseLabelOutline(labelmapConfig.renderOutline);
+  volumeActor.getProperty().setUseLabelOutline(renderOutline);
+
+  // @ts-ignore: setLabelOutlineWidth is not in the vtk.d.ts appraently
```
Comment (swederik) [inline packages/tools/src/tools/displayTools/Labelmap/labelmapDisplay.ts:297]: Typo

File: packages/tools/src/tools/displayTools/Labelmap/labelmapDisplay.ts [hunk 9]
```diff
@@ -339,36 +448,48 @@ function _removeLabelmapFromToolGroupViewports(
...
-    );
-  }
+  await addLabelmapToElement(
+    viewport.element,
+    volumeId,
+    segmentationRepresentationUID
+  );
 }
 
+// async function _addLabelmapToToolGroupViewports(
```
Comment (swederik) [inline packages/tools/src/tools/displayTools/Labelmap/labelmapDisplay.ts:463]: Remove it

**Review by swederik — APPROVED** [review]
Looks fine just minor comments

File: packages/tools/src/stateManagement/segmentation/removeSegmentationsFromToolGroup.ts (hunk 3 above)
Comment (sedghi) [reply packages/tools/src/stateManagement/segmentation/removeSegmentationsFromToolGroup.ts:56]: If one is removing couple of 10 segmentations and don't want to render after each?

**Pull request merged**